tests/unit/test_main.py

```diff
@@ -1,14 +1,8 @@
 import pytest
 
 from src.main import (
-    hello,
     main,
 )
 
-def test_hello():
-    expected = "Hello world!"
-    actual = hello()
-    assert actual == expected
-
 def test_main():
     main()
```

Comment:
This was just a placeholder mb, but our `main` function is a function just like the rest of our implementation! That means, we also should unit test this function. As with any other function, we want to make sure our `main` function is doing the things we expect. So in this case, `main` is calling the 5 diff helper functions, so what we can do is **assert** that it's calling the functions as we expect. The way we'd go about this is [we mock our helper functions (they become `MagicMock`s)](https://docs.python.org/3/library/unittest.mock.html), call `main`, and then [assert that our mocks were called](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_once_with). For example:
```python
from unittest.mock import patch

...
# When you patch, it get filled into the params in reverse order (don't ask me why).
@patch("src.main.related_movies")
@patch("src.main.actor_images")
@patch("src.main.actors")
@patch("src.main.extract_movie_data")
@patch("src.main.toprated_movies")
def test_main(mock_toprated_movies, mock_extract_movie_data, mock_actors, mock_actor_images, mock_related_movies):
    main()

    mock_toprated_movies.assert_called_once()
    mock_extract_movie_data.assert_called_once()
    mock_actors.assert_called_once()
    mock_actor_images.assert_called_once()
    mock_related_movies.assert_called_once()
```
There's two other things that are nice about patching/`MagicMock`s:
- you can also make sure it's being called with the arguments you expect,
- you can also make your mocks return a value you specify,

which when you combine these, you can do something like this:

```python
def test_main(mock_top_rated_movies, ...):
    mock_top_rated_movies.return_value = "mock-data-1"
    mock_extract_movie_data.return_value = "mock-data-2"
    ...
    
    expected_headers = {"accept": ...}
    main()

    mock_top_rated_movies.assert_called_once_with(4, expected_headers)
    mock_extract_movie_data.assert_called_once_with("mock-data-1")
    mock_actors.assert_called_once_with("mock-data-2", expected_headers)
    ...
```